packages/contracts-bedrock/src/L1/OptimismPortal.sol

```diff
@@ -384,13 +541,37 @@ contract OptimismPortal is Initializable, ResourceMetering, ISemver {
         // Compute the opaque data that will be emitted as part of the TransactionDeposited event.
         // We use opaque data so that we can update the TransactionDeposited event in the future
         // without breaking the current interface.
-        bytes memory opaqueData = abi.encodePacked(msg.value, _value, _gasLimit, _isCreation, _data);
+        bytes memory opaqueData = abi.encodePacked(_mint, _value, _gasLimit, _isCreation, _data);
 
         // Emit a TransactionDeposited event so that the rollup node can derive a deposit
         // transaction for this deposit.
         emit TransactionDeposited(from, _to, DEPOSIT_VERSION, opaqueData);
     }
 
+    /// @notice Sets the gas paying token for the L2 system. This token is used as the
+    ///         L2 native asset. Only the SystemConfig contract can call this function.
+    function setGasPayingToken(address _token, uint8 _decimals, bytes32 _name, bytes32 _symbol) external {
+        if (msg.sender != address(systemConfig)) revert Unauthorized();
+
+        // Set L2 deposit gas as used without paying burning gas. Ensures that desposits cannot use too much L2 gas.
+        useGas(80000);
```

Comment:
I don't think this needs to live in constants as its used once

Reply:
IIUC Shouldn't it match a value in the resource config?

Reply:
> IIUC Shouldn't it match a value in the resource config?

Please specifically describe which value it should match. This is not expressing "buy all gas". Its simply using enough to execute on L2

Reply:
OK, I see now, the comment is helpful.

What do you think about a test to ensure that this value is/remains sufficient?

Reply:
The amount of gas being sufficient is covered in the op-e2e tests. It will not be super accurate to do it in solidity. The L1 attributes tx [uses](https://basescan.org/tx/0xa192af61c0a7401d3ae7a8838f2e274f5e0925dd8fa165ed7e7232da592142d1) 43k gas and stores more data

Reply:
I do not think there is a need for a solidity test for this but am open to other opinions